extensions/configoperations/configoperations.go

```diff
@@ -0,0 +1,95 @@
+package configoperations
+
+import (
+	"errors"
+	"fmt"
+
+	"sigs.k8s.io/yaml"
+)
+
+// ReplaceValue iterates through a map to a provided key path and replaces the value.
+func ReplaceValue(keyPath []string, replaceVal any, searchMap map[string]any) (map[string]any, error) {
+	if len(keyPath) <= 1 {
+		searchMap[keyPath[0]] = replaceVal
+
+		return searchMap, nil
+	} else {
+		var err error
+
+		if _, ok := searchMap[keyPath[0]].(map[string]any); ok {
+			searchMap[keyPath[0]], err = ReplaceValue(keyPath[1:], replaceVal, searchMap[keyPath[0]].(map[string]any))
+			if err != nil {
+				return nil, err
+			}
+		} else if _, ok := searchMap[keyPath[0]].([]any); ok {
+			for i := range searchMap[keyPath[0]].([]any) {
+				searchMap[keyPath[0]].([]any)[i], err = ReplaceValue(keyPath[1:], replaceVal, searchMap[keyPath[0]].([]any)[i].(map[string]any))
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
+	}
+
+	return searchMap, nil
+}
+
+// GetValue iterates through a map to a provided key path and returns the value.
+func GetValue(keyPath []string, searchMap map[string]any) (any, error) {
+	var err error
+	var keypathvalues any
+	if len(keyPath) == 1 {
+		keypathvalues, ok := searchMap[keyPath[0]]
+		if !ok {
+			err = errors.New(fmt.Sprintf("expected key does not exist: %s", keyPath[0]))
+		}
+		return keypathvalues, err
+	} else {
+		if _, ok := searchMap[keyPath[0]].(map[string]any); ok {
+			keypathvalues, err = GetValue(keyPath[1:], searchMap[keyPath[0]].(map[string]any))
+			if err != nil {
+				return nil, err
+			}
+		} else if _, ok := searchMap[keyPath[0]].([]any); ok {
+			for i := range searchMap[keyPath[0]].([]any) {
+				keypathvalues, err = GetValue(keyPath[1:], searchMap[keyPath[0]].([]any)[i].(map[string]any))
```

Comment:
does keypathvalues not need some [i] value here since its in a for loop?

Reply:
* I'll get some jenkins runs for this up. I have it hooked up to RKE2DynamicInput in my test branch

* Sure, I could add a readme to Permutations with instructions on how to utilize it. Or would you just prefer a code example in rancher/rancher?

Reply:
For the keypathvalues in the case that we hit a list of maps I made the choice to have this as the default behavior. 

The reason for this is that if we have a list of maps we run into a scenario where GetValue/ReplaceValue have multiple values to get or replace and would have to choose which ones to perform that operation on. An example of this would be if we wanted to get/replace the ami field and we had multiple machineConfigs, which ami field should we get/replace?

 Currently I don't support that in permutations and frankly there isn't really a scenario where we need to right now.(as far as i know) . If/When there is a reason to update this I will do so however right now only getting/replacing the first value is sufficient.

Reply:
Godoc supports examples, in my opinion, the function examples should live where the code lives (maybe not necessarily config-related but how it works and how the output would look). You can also put an example here for the functions, there should be a comment example in v1 clients for converting

Reply:
what about custom cluster for windows ami? 

```
Or would you just prefer a code example in rancher/rancher?
```
I think in rancher is fine

Reply:
Added a godoc example to the permutations package